File: cmd/debug.go
package cmd

import (
	"fmt"

	"github.com/mturley/agent-handler/config"
	"github.com/spf13/cobra"
)

var debugCmd = &cobra.Command{
	Use:   "debug [enable|disable]",
	Short: "Toggle debug info in the statusline",
	Args:  cobra.ExactArgs(1),
	RunE:  runDebug,
}

func init() {
	debugCmd.GroupID = "admin"
	rootCmd.AddCommand(debugCmd)
}

func runDebug(cmd *cobra.Command, args []string) error {
	action := args[0]
	if action != "enable" && action != "disable" {
		return fmt.Errorf("usage: handler debug [enable|disable]")
	}

	cfgPath := config.DefaultPath()
	cfg, err := config.Read(cfgPath)
	if err != nil {
		cfg = &config.Config{}
	}

	cfg.Debug = action == "enable"

	if err := config.Write(cfgPath, cfg); err != nil {
		return fmt.Errorf("failed to write config: %w", err)
	}

	if cfg.Debug {
		fmt.Println("✓ Debug mode enabled — statusline will show debug info for all sessions")
	} else {
		fmt.Println("✓ Debug mode disabled")
	}
	return nil
}

File: src/Handler/Cmd/Debug.hs
-- | Port of cmd/debug.go: toggle debug info in the statusline.
module Handler.Cmd.Debug (debugCommand) where

import Data.Text (Text)
import Options.Applicative

import Handler.Cli.Common
import Handler.Config (Config(..), configDefaultPath, readConfig, writeConfig)

debugCommand :: Mod CommandFields NamedCommand
debugCommand = mkCommand "debug" "Toggle debug info in the statusline"
  (runDebug <$> strArgument (metavar "enable|disable"))

runDebug :: Text -> Ctx -> IO ()
runDebug action _
  | action /= "enable" && action /= "disable" =
      dieText "usage: handler debug [enable|disable]"
  | otherwise = do
      cfgPath <- configDefaultPath
      cfg <- readConfig cfgPath
      let cfg' = cfg { debug = action == "enable" }
      writeConfig cfgPath cfg'
      putTextLn $ if cfg'.debug
        then "✓ Debug mode enabled — statusline will show debug info for all sessions"
        else "✓ Debug mode disabled"
